src/openrct2/localisation/Formatter.h

```diff
@@ -77,16 +77,16 @@ class Formatter
 
         // clang-format off
         static_assert(
-            std::is_same_v<TSpecified, char*> ||
-            std::is_same_v<TSpecified, const char*> ||
-            std::is_same_v<TSpecified, int16_t> ||
-            std::is_same_v<TSpecified, int32_t> ||
-            std::is_same_v<TSpecified, money32> ||
-            std::is_same_v<TSpecified, rct_string_id> ||
-            std::is_same_v<TSpecified, uint16_t> ||
-            std::is_same_v<TSpecified, uint32_t> ||
-            std::is_same_v<TSpecified, utf8*> ||
-            std::is_same_v<TSpecified, const utf8*>
+            std::is_same_v<typename std::remove_cv<TSpecified>::type, char*> ||
```

Comment:
You could make this a lot cleaner by using
```cpp
using TCVType = typename std::remove_cv_t<TSpecified>;
```
Above the assert.

Reply:
Not sure if that's really cleaner. That removes the multiple calls to `std::remove_cv`, however, unless you go with a verbose identifier, it isn't obvious which types it actually is the base type of. `TSpecified` and `TDeduced` can be different. Since it's all computed at compile time, I don't see a reason to make it less obvious.

If you want to avoid the duplicated uses of `std::remove_cv`, perhaps a better approach to checking if the type is any of a given list. This could be achieved with a variadic template.

```cpp
template<class T, class... Rest>
inline constexpr bool is_any = (std::is_same_v<T, Rest> || ...);

...

static_assert(
    is_any<
        typename std::remove_cv<TSpecified>::type, char*, const char*, int16_t, int32_t, money32, rct_string_id, uint16_t, uint32_t, utf8*,
        const utf8*>,
    "Specified type is not supported");
```

And while this is compacter and has fewer duplicated lines, it is less readable.